src/xcoobee/sdk/Consents.js

```diff
@@ -519,6 +519,70 @@ class Consents {
     }
   }
 
+  /**
+   * Register consents
+   *
+   * @param {Array<{ target: string, date_received: ?string, date_expires: ?string }>} targets
+   * @param {?string} [reference]
+   * @param {?string} [filename]
+   * @param {?string} [campaignId]
+   * @param {Config} [config]
+   * @returns {Promise<SuccessResponse>}
+   */
+  async registerConsents(targets, reference = null, filename = null, campaignId = null, config = null) {
+    this._assertValidState();
+    const resolvedCampaignId = SdkUtils.resolveCampaignId(campaignId, config, this._.config);
+    const sdkCfg = SdkUtils.resolveSdkCfg(config, this._.config);
+    const {
+      apiKey,
+      apiSecret,
+      apiUrlRoot,
+    } = sdkCfg;
+
+    const errors = [];
+    const progress = [];
+    let response;
+
+    try {
+      const apiAccessToken = await this._.apiAccessTokenCache.get(apiUrlRoot, apiKey, apiSecret);
+      const user = await this._.usersCache.get(apiUrlRoot, apiKey, apiSecret);
+      const userCursor = user.cursor;
+
+      const successfullyUploadedFiles = [];
+      const result = { progress, ref_id: null };
+      if (filename) {
+        const endPointName = UploadPolicyIntents.OUTBOX;
+        const fileUploadResults = await FileUtils.upload(apiUrlRoot, apiAccessToken, userCursor, endPointName, [filename]);
+
+        fileUploadResults.forEach((fileUploadResult) => {
+          const { error, file, success } = fileUploadResult;
+          if (success) {
+            successfullyUploadedFiles.push(file);
+            progress.push(`successfully uploaded ${file}`);
+          } else {
+            errors.push(`Failed to upload file: ${file}. Error: ${error}.`);
+            progress.push(`failed upload on ${file}`);
+          }
+        });
+      }
+
+      if (!errors.length) {
+        const refId = await ConsentsApi.registerConsents(apiUrlRoot, apiAccessToken, resolvedCampaignId, targets, reference, successfullyUploadedFiles.length > 0 ? filename : null);

```

Comment:
`successfullyUploadedFiles.length > 0 ? filename : null` get rid of this condition. 
[here](https://github.com/XcooBee/xcoobee-js-sdk/pull/28/files#diff-5becabcaab0e2be81dc4047a5718c569R563) throw an error, that file is not uploaded